src/components/checklist/CheckItem.tsx

```diff
@@ -1,15 +1,73 @@
-import CheckItem from './CheckItem';
+import { CheckItemProps } from './type';
+import { useEffect, useRef, useState } from 'react';
+import Dropdown from './Dropdown';
+
+const CheckItem = ({ id, content, isDone, onIsDoneChange, onContentChange, onDeleteClick }: CheckItemProps) => {
+  const [isMenuHidden, setIsMenuHidden] = useState(true);
+  const [inputValue, setInputValue] = useState(content);
+  const [isEdit, setIsEdit] = useState(inputValue === '' ? true : false);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  /* item 수정해야 하는 상황에선 바로 포커스 */
+  useEffect(() => {
+    if (isEdit) {
+      inputRef.current?.focus();
+    }
+  }, [isEdit]);
+
+  /* 함수 이름 전반적으로 정리 예정 */
+  const handleCheckItemContentUpdate = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === 'Enter') {
+      setIsEdit(false);
+      onContentChange(id, inputValue);
+    }
+  };
+
+  const handleCheckItemIsDoneUpdate = () => {
+    onIsDoneChange(id);
+  };
+
+  const handleInputValueUpdate = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const target = e.target as HTMLInputElement;
+    setInputValue(target.value);
+  };
+
+  const handleMenuDropdown = () => {
+    setIsMenuHidden(!isMenuHidden);
```

Comment:
```suggestion
    setIsMenuHidden(prev => !prev);
```

이런식으로도 사용이 가능해요!
스코프에 있는 값을 참조를 해도 되긴 하는데,  다른 이벤트에서 같은 상태를 변경하는 경우가 생길수도 있고해서 함수형으로 업데이트하는걸 권장하는 걸로 알고 있어요:)